spanner/spannertest/db.go

```diff
@@ -73,6 +75,20 @@ type colInfo struct {
 	Alias    spansql.PathExp // an alternate name for this column (result sets only)
 }
 
+type index struct {
```

Comment:
noting that an important aspect of spannertest is that it is _not_ intended to be performant, the only real aspect of correctness with regard to indexes (aside from tracking which ones exist, etc.) is that partial indexes (e.g. NULL_FILTERED) should return a subset of values when the index is used for reads. Other than that, there doesn't need to be any substantial use of indexes, nor any efficiency-oriented data structures like the btree.

Perhaps most of this should be dropped, and only track the index construction, then modify the read code so it can simulate the row skipping when using a partial index?

Reply:
I mean, spannertest is explicitly not for performance, so if there isn't anything like null filtering (where you get different response values if you use such an index for reading) then there isn't much point in actually maintaining an index.